### PR TITLE
Transfer assigned memberships to a special account

### LINKED ISSUE #8: Send assigned memeberships to a keyless account instead of burning
When a membership is assigned, it's burned from the account of the group in the special "collection 0" and re-minted in a collection specific to the group that is assigning a new member. This comes with the issue that collection 0 *forgets* it ever had a given membership allowing for a membership with a given id to be re-minted. We want membership ids to remain unique across all collections, we can do that if instead of burning we transfer the membership to an account not controlled by anyone.

### PR DESCRIPTION
**Pull request by pandres95**
Closes #8

### CODE REVIEW

File: traits/memberships/src/impl_nonfungibles.rs [hunk 1]
```diff
@@ -66,7 +71,12 @@ where
         Self::set_rank(group, m, 0)?;
         T::burn(group, m, None)?;
         let count = Self::members_total(group);
-        T::set_typed_collection_attribute(group, &ATTR_MEMBER_TOTAL, &(count - 1))
+        T::set_typed_collection_attribute(group, &ATTR_MEMBER_TOTAL, &(count - 1))?;
+
+        let mgr_group = Self::Group::zero();
+        let group_owner = T::collection_owner(group)
+            .expect("the group existed when burning the membership, the group has an owner; qed");
```
Comment (olanod) [inline traits/memberships/src/impl_nonfungibles.rs:78]:
```suggestion
            .expect("the group existed when assigning the membership, the group has an owner; qed");
```

File: traits/memberships/src/impl_nonfungibles.rs [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+    pallet_prelude::DispatchError,
+    sp_runtime::{str_array, traits::Zero},
     traits::tokens::nonfungibles_v2 as nonfungibles,
 };
 
 const ATTR_MEMBER_TOTAL: &[u8] = b"membership_member_total";
 const ATTR_MEMBER_RANK: &[u8] = b"membership_member_rank";
 const ATTR_MEMBER_RANK_TOTAL: &[u8] = b"membership_rank_total";
 
+pub const ASSIGNED_MEMBERSHIPS_ACCOUNT: [u8; 32] = str_array("modlkv/mbrsh_reserved");
```
Comment (olanod) [inline traits/memberships/src/impl_nonfungibles.rs:12]:
```suggestion
pub const ASSIGNED_MEMBERSHIPS_ACCOUNT: [u8; 32] = str_array("memberships/assigned_memberships");
```
`mod` is for pallets and `kv` kreivo and this implementation is neither :)

**Review by olanod — APPROVED** [review]

File: traits/memberships/src/impl_nonfungibles.rs [hunk 3]
```diff
@@ -66,7 +71,12 @@ where
         Self::set_rank(group, m, 0)?;
         T::burn(group, m, None)?;
```
Comment (pandres95) [inline traits/memberships/src/impl_nonfungibles.rs:72]: @olanod here's the burning

File: traits/memberships/src/impl_nonfungibles.rs (hunk 1 above)
Comment (pandres95) [reply traits/memberships/src/impl_nonfungibles.rs:78]: Actually, it'd be burning.

Notice that, before this assertion, we're burning the released membership from the group we're trying to get it's owner from.

**Pull request merged**